src/log/ContextProcessor.php

```diff
@@ -84,18 +85,27 @@ public function __invoke(array $record): array
                 // NBD
             }
 
-            $record[$this->key]['body'] = $body;
+            $data['body'] = $body;
         }
 
+        $message = null;
         if ($vars = $this->filterVars($this->vars)) {
             if ($this->dumpVars) {
-                $record['message'] .= "\n" . $this->dumpVars($vars);
+                $message = "\n" . $this->dumpVars($vars);
             } else {
-                $record[$this->key]['vars'] = $vars;
+                $data['vars'] = $vars;
             }
         }
 
-        return $record;
+        return new LogRecord(
+            datetime: $record->datetime,
+            channel: $record->channel,
+            level: $record->level,
+            message: $record->message . $message,
+            context: ($this->key === 'context') ? $record->context + $data : $record->context,
+            extra: ($this->key === 'extra') ? $record->extra + $data : $record->extra,
```

Comment:
This certainly should work…but I'm not seeing (nor do I recall) a reason for `\craft\log\ContextProcessor::key` to exist (seems like its just always `context`).

 I'm tempted to say just drop that prop, and just set `LogRecord::context`.

Reply:
I couldn’t figure out what was the use of it either. We don’t set that `$key` anywhere, and it always seems to be `context`. I did some searching and reading, which didn’t seem to give any hints. Since you also think this might not be used, I dropped it.